commons/src/main/kotlin/com/simplemobiletools/commons/dialogs/WritePermissionDialog.kt

```diff
@@ -11,39 +11,39 @@ import kotlinx.android.synthetic.main.dialog_write_permission.view.*
 import kotlinx.android.synthetic.main.dialog_write_permission_otg.view.*
 
 class WritePermissionDialog(activity: Activity, val mode: Mode, val callback: () -> Unit) {
-    enum class Mode {
-        OTG,
-        SD_CARD,
-        OPEN_DOCUMENT_TREE_SDK_30,
-        CREATE_DOCUMENT_SDK_30,
+    sealed class Mode {
+        object Otg : Mode()
+        object SdCard : Mode()
+        data class OpenDocumentTreeSDK30(val path: String) : Mode()
+        object CreateDocumentSDK30 : Mode()
```

Comment:
what are these sealed, object etc? Looks weird

Reply:
is it supposed to be some enums?

Reply:
Only the OpenDocumentTreeSDK30 case needs a path. 
I didn't want to pass something like a null or an empty string for the other cases.